### ISSUE TITLE
Add support for __float128 conversions

### ISSUE THREAD

**Issue by benruijl** [advanpix/mpreal#6]
Starting from  MPR 4.0, the functions `mpfr_set_float128` and `mpfr_get_float128` can be used to convert from and to `__f128` (see test cases here for usage: https://fossies.org/linux/misc/mpfr-4.1.0.tar.gz/mpfr-4.1.0/tests/tset_float128.c).

It would be great if `mpreal` supports these conversions as well.

**Comment by benruijl** [comment]
In my forked version https://github.com/benruijl/mpreal/commit/9cf67d12ec23a8e8ce117cb9194dac57b3dcd674 I've added support for `__float128` by essentially copying all `long double` functions. These functions may all need to be wrapped in `#if` clauses to support platforms that don't have `__float128` (I'm also very open to volunteers, as this is not currently a high priority for me). Once this is done, I can make a pull request.

**Comment by maxale** [comment]
Support for `__int128` is also missing.

**Comment by advanpix** [comment]
> Support for `__int128` is also missing.

MPFR has no direct support for this GCC-specific extension (__int128) as this is an integer type, and out of scope of MPFR functionality. GMP/MPIR would be a better choice I guess. Or you can add the conversion to/from __int128 if you need this specific functionality.
